ls1intum/Artemis
# [Bugfix] Allowing Editors to trigger Instructor Builds

### Checklist

### Motivation and Context
At the moment, editors cannot trigger the instructor build for template and solution participation in programming exercises, even though they should be able to do it and have access to the respective button on the exercise detail page.

### Description
I have added a new check that makes sure that editors can issue instructor builds for template and solution participation, but not for student participation.

### Steps for Testing

**You will need:** 1x TA, 1x Editor, 1x instructor, 1x Programming Exercise, 1x Student Submission on this Exercise

1. With the TA account: Go to the programming exercise detail view. Make sure that the resubmit buttons for template and solution participation are not present.
2. Repeat step 1. with the Editor account and make sure that the buttons are now present.
3. With the Editor, click on both buttons and make sure that the participation status changes to "Building and testing"
4. Wait for the result and make sure that a result is actually being generated.
5. Open the template and solution build plan on Bamboo and make sure that the build was actually triggered
6. With the Editor, go to the Participation View of the programming exercise. Make sure that no resubmit button is present.
7. Repeat step 6. with the Instructor. Make sure that the button is present and actually works.
